### PR TITLE
RABBIT: init at 1.0.0

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- Add `RABBIT`: https://github.com/natarajan-chidambaram/RABBIT

CC: @natarajan-chidambaram

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by AndersonTorres — COMMENTED** [review]
Is there any reason to name such package in *uppercase* -- besides following upstream?

**Comment by drupol** [comment]
> Is there any reason to name such package in UPPERCASE -- besides following upstream?

No.

**Comment by AndersonTorres** [comment]
Well, let's keep  with the unwritten convention, then.
As far as I remember, only SDL and friends are treated as exceptions.

**Review by AndersonTorres — COMMENTED** [review]
Result of `nixpkgs-review pr 300209` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>RABBIT</li>
    <li>RABBIT.dist</li>
  </ul>
</details>

Tested the "binary", it shows a typical help text.

LGTM (except the *uppercase* name).

**Comment by drupol** [comment]
@AndersonTorres I updated the name, setting it to lowercase as you suggested.

Let me know if it's OK for you to merge it.

File: pkgs/by-name/ra/rabbit/package.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  pythonRelaxDeps = true;
+
+  nativeBuildInputs = [
+    python3.pkgs.setuptools
+    python3.pkgs.wheel
+    python3.pkgs.pythonRelaxDepsHook
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
```
Comment (AndersonTorres) [inline pkgs/by-name/ra/rabbit/package.nix:26]: I am not a Python expert, but nowadays this is written as
```suggestion
  dependencies = with python3.pkgs; [
```

File: pkgs/by-name/ra/rabbit/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  src = fetchFromGitHub {
+    owner = "natarajan-chidambaram";
+    repo = "RABBIT";
+    rev = version;
+    hash = "sha256-IRG3OcWutkZA4oegeEIDyaIadl3dLaMneqOt/H2/Il4=";
+  };
+
+  pythonRelaxDeps = true;
+
+  nativeBuildInputs = [
```
Comment (AndersonTorres) [inline pkgs/by-name/ra/rabbit/package.nix:20]: ... and this as

```suggestion
  build-system = [
```

**Review by AndersonTorres — COMMENTED** [review]
It should work fine. LGTM.

File: pkgs/by-name/ra/rabbit/package.nix (hunk 1 above)
Comment (drupol) [reply pkgs/by-name/ra/rabbit/package.nix:26]: Ho first time I see this. Curious to learn more about it!

File: pkgs/by-name/ra/rabbit/package.nix (hunk 2 above)
Comment (drupol) [reply pkgs/by-name/ra/rabbit/package.nix:20]: First time too !

File: pkgs/by-name/ra/rabbit/package.nix (hunk 1 above)
Comment (drupol) [reply pkgs/by-name/ra/rabbit/package.nix]: Done, thank you, TIL !

File: pkgs/by-name/ra/rabbit/package.nix (hunk 2 above)
Comment (drupol) [reply pkgs/by-name/ra/rabbit/package.nix]: Done, thank you, TIL !

**Pull request merged**